### PR TITLE
KZOO-409: Two recordings generated for a single call

### PR DESCRIPTION
**Pull request by Navoda626**
Previously, recordings were done in loopback legs which are eventually going to bow-out, and stored in the database.

With this change (check whether the recording stop event is from a loopback and bow-out leg and skip storing) now, recordings done in loopback legs which bowed-out will skip storing in the database.

Related PR: https://github.com/2600hz/kazoo-core/pull/794

### CODE REVIEW

File: src/media_recording.erl [hunk 1]
```diff
@@ -477,18 +487,18 @@ recording_db(Data, DefaultDB) ->
...
 -spec should_store_recording(kz_json:object(), kz_term:api_ne_binary(), kz_term:api_ne_binary()) -> boolean().
-should_store_recording(_Data, undefined, _Url) -> false;
+should_store_recording(_Data, 'undefined', _Url) -> 'false';
 should_store_recording(Data, AccountId, Url) ->
     case kz_json:get_boolean_value(<<"should_store_recording">>, Data) of
-        undefined -> kapps_call_recording:should_store_recording(AccountId, Url);
-        true -> {'true', 'local'};
-        false -> false
+        'undefined' -> kapps_call_recording:should_store_recording(AccountId, Url);
+        'true' -> {'true', 'local'};
```
Comment (wdhammika) [inline src/media_recording.erl:495]: The function return and the return type of the function specification don't match. Please check it.

**Comment by lazedo** [comment]
this is the wrong fix, if you really want to implement this, it should never start the recording, that's what you need to implement

**Comment by Navoda626** [comment]
@lazedo I was unable to find a way to stop recordings for loopback bow out legs. Can you please guide me to find a way we can identify in kazoo whether the recording leg is a loopback bow-out leg before the start of the recording?

**Comment by k-anderson** [comment]
@lazedo the issue was the only check prior to starting the recording would be to see if bowout was enabled, but thats not a guaranty that the loopback will ACTUALLY do a bow out.  I talked to Navoda who started down that path and was having issues, and on that call concluded that it was safer to start the recording then make the decision based on the result on termination.  Do you have any specific implementation thoughts how it should be done prior to starting?

**Comment by lazedo** [comment]
@k-anderson it shouldn't start recording on cfwd , uri, cell. if enabled for recording, it should be done on the cf_resources when the `real` outbound leg is created, ice, it passes info along to start `on answer`

**Comment by Navoda626** [comment]
Tested this in dev env with call forward scenarios and this change works

**Pull request merged**